### PR TITLE
Move and relocate 'Frame Delay'

### PR DESCRIPTION
**Pull request by sonninnos**
## Description

While developing the "Frame Rest" feature I accidentally noticed that it also affected input response, and not only CPU usage. And with closer inspection I noticed that the current location of "Frame Delay" is utterly useless when it is done before `core_run()` and not after. Sure, feels counterintuitive at first, but the test results speak for themselves, and in the grand scheme loop the sleeping still happens and works as advertised. What exactly makes this massive difference, I can not say, but I'd like to know.

The simplest test for confirming this is using a lightweight core such as Genesis Plus GX with a lightgun device and crosshair enabled, and then showing OS mouse cursor by ungrabbing mouse, aligning the cursors by pushing them to a corner, and observing the delay. It is extremely immediate and lagless this way, especially when pushed above the default average frame time, and of course with all the other video driver specific settings optimized.

If someone can demonstrate that with some platform it is actually worse now, we'll make it an option. With Windows every video driver behaves exactly the same. I moved all the code to `video_driver.c` while at it to keep the runloop clean.

Basically this change makes "Frame Rest" redundant, but let's let it stay for now, and decide later if "Frame Delay" should also be allowed to run when menu is open, and how they should eat each other when both are enabled.

### CODE REVIEW

**Comment by crashGG** [comment]
The way I check the total input/output delay is to set the keyboard Caps Lock key to fire, then use a high-speed camera (120fps) to shoot the keyboard LED indicator and the character firing in game on the screen into the same video, and then Calculate the number of frames delay in playback  during playback .
Maybe this immature method can provide you with some reference.

**Comment by sonninnos** [comment]
Sure, I've been doing that too with a 60fps camera, and it shows a difference.
